src/Core/src/Platform/iOS/SwitchExtensions.cs

```diff
@@ -17,17 +17,20 @@ public static void UpdateTrackColor(this UISwitch uiSwitch, ISwitch view)
 			if (view == null)
 				return;
 
-			if (view.TrackColor != null)
-				uiSwitch.OnTintColor = view.TrackColor.ToPlatform();
+			var uIView = GetTrackSubview(uiSwitch);
 
-			UIView uIView;
-			if (OperatingSystem.IsIOSVersionAtLeast(13) || OperatingSystem.IsTvOSVersionAtLeast(13))
-				uIView = uiSwitch.Subviews[0].Subviews[0];
-			else
-				uIView = uiSwitch.Subviews[0].Subviews[0].Subviews[0];
+			if (!view.IsOn)
+			{
+				// iOS 13+ uses the UIColor.SecondarySystemFill to support Light and Dark mode
+				// else, use the RGBA equivalent of UIColor.SecondarySystemFill in Light mode
+				uIView.BackgroundColor = OperatingSystem.IsIOSVersionAtLeast(13) ? UIColor.SecondarySystemFill : UIColor.FromRGBA(120, 120, 128, 40);
+			}
 
-			if (view.TrackColor != null)
+			else if (view.TrackColor is not null)
```

Comment:
if you have if, else if without an else, that else if is in reality an if, for example:
```cs
if (condition) {
} else if (condition2) {
}
```
is the same to the more elegant:
```cs
if (condition) {
}
if (condition2) {
}
```

Reply:
> if you have if, else if without an else, that else if is in reality an if, for example:
> 
> ```cs
> if (condition) {
> } else if (condition2) {
> }
> ```
> 
> is the same to the more elegant:
> 
> ```cs
> if (condition) {
> }
> if (condition2) {
> }
> ```

Those are not equivalent. In the first example, at _most_ one of those code blocks can run. In the second, _both_ of them can run if both `condition` and `condition2` are true.

And also TJ's point about not evaluating `condition2`.